### PR TITLE
Add clone with changes to python client

### PR DESCRIPTION
**Pull request by syandroo**

### CODE REVIEW

File: launch/client.py [hunk 1]
```diff
@@ -812,6 +812,34 @@ def get_model_bundle(
         ), f"Bundle with name `{bundle_name}` not found"
         return ModelBundle.from_dict(resp["bundles"][0])  # type: ignore
 
+    def clone_model_bundle_with_changes(
+        self,
+        existing_bundle: Union[ModelBundle, str],
+        new_bundle_name: str,
+        app_config: Dict[str, str],
```
Comment (seanshi-scale) [inline launch/client.py:819]: should this be optional? (i.e. if it's None or something we use the old app_config, otherwise we use the new one)

**Review by seanshi-scale — APPROVED** [review]
lgtm if it's been tested, also had a comment about making `app_config` optional but idk what we quite want to do here

**Review by chongzluong — APPROVED** [review]
Assuming u tested this 👀

File: launch/client.py (hunk 1 above)
Comment (syandroo) [reply launch/client.py:819]: hmmm i think setting it to null makes more sense?

**Comment by syandroo** [comment]
tested that it works - server was throwing an error when it didn't need to tho, will fix server side

**Pull request merged**